components/server/src/auth/subject-id.ts

```diff
@@ -73,7 +73,7 @@ export class SubjectId {
 /**
  * Interface type meant for backwards compatibility
  */
-export type Subject = string | SubjectId;
+export type Subject = string | SubjectId | undefined;
```

Comment:
do we really want to include undefined in this type? I find that a bit odd, or is it a leftover?

Reply:
It's odd, yes, that's why I waited for so long to add it.
I tried without it, but it's necessary to split up the work around removing the "first argument"; otherwise we end up with almost a single/big PR. :grimacing:

Reply:
Ah, ok, as long as we are going to remove it eventually again it's fine

Reply:
Yeah, totally. :100: